galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/routes/AddPropertyInNamespaceRoute.java

```diff
@@ -0,0 +1,125 @@
+import dev.galasa.framework.api.common.resources.CPSFacade;
+import dev.galasa.framework.api.common.resources.CPSNamespace;
+import dev.galasa.framework.api.common.resources.CPSProperty;
+import dev.galasa.framework.spi.ConfigurationPropertyStoreException;
+import dev.galasa.framework.spi.FrameworkException;
+import dev.galasa.framework.spi.IFramework;
+
+public class AddPropertyInNamespaceRoute extends CPSRoute {
+
+    protected static final String path = "\\/namespace/([a-z][a-z0-9]+)/property/([a-zA-Z0-9\\.\\-\\_]+)/?";
+    
+    private String propertyName;
+    private String namespaceName;
+    
+    public AddPropertyInNamespaceRoute(ResponseBuilder responseBuilder, IFramework framework) {
+        /* Regex to match endpoints: 
+		*  -> /cps/namespace/namespaceName//property/propertyName
+		*  -> /cps/namespace/namespaceName//property/propertyName/
+		*/
+        super(responseBuilder, path, framework);
+    }
+    
+    @Override
+    public HttpServletResponse handlePutRequest(String pathInfo, QueryParameters queryParams,HttpServletRequest req, HttpServletResponse response)
+            throws ServletException, FrameworkException, IOException {
+        getPropertyDetailsFromURL(pathInfo);
+        checkNamespaceExists(namespaceName);
+        checkRequestHasContent(req);
+        ServletInputStream body = req.getInputStream();
+        String jsonString = new String (body.readAllBytes(),StandardCharsets.UTF_8);
+        body.close();
+        JsonObject reqJson = gson.fromJson(jsonString,JsonObject.class);
+        String properties = setNamespaceProperty(reqJson);
+        return getResponseBuilder().buildResponse(req, response, "application/json", properties, HttpServletResponse.SC_OK); 
+    }
+
+    private void getPropertyDetailsFromURL(String pathInfo) throws InternalServletException {
+        // Set the values for the suffix prefix and namespaceName in order to use the values
+        try {
+            String[] path = pathInfo.split("/");
+            namespaceName = path[2];
+            propertyName = path[4];
+        } catch (Exception e) {
+            ServletError error = new ServletError(GAL5000_GENERIC_API_ERROR);  
+            throw new InternalServletException(error, HttpServletResponse.SC_NOT_FOUND, e);
+        }
+    }
+
+    private String setNamespaceProperty(JsonObject reqJson) throws InternalServletException, IOException{
+        String properties = "";
+         try {
+            nameValidator.assertNamespaceCharPatternIsValid(namespaceName);
+            CPSFacade cps = new CPSFacade(framework);
+            CPSNamespace namespace = cps.getNamespace(namespaceName);
+            if (namespace.isHidden()) {
+                ServletError error = new ServletError(GAL5016_INVALID_NAMESPACE_ERROR, namespaceName);
+                throw new InternalServletException(error, HttpServletResponse.SC_NOT_FOUND);
+            }
+            if (!propertyName.equals(reqJson.get("name").getAsString())){
```

Comment:
Minor - we could define variables for `name` and `value` instead of repeating `reqJson.get(...).getAsString()` everywhere